src/ServicePulse.Host/app/js/views/endpoint_details/endpoint_details.controller.js

```diff
@@ -45,6 +45,15 @@
                     $scope.loading = false;
                 }
 
+                $scope.throughput = $scope.endpoint.graphData.throughput;
+                $scope.throughput.xAxisPoints = $scope.endpoint.graphData.timeAxisValues;
+
+                $scope.throughput.xAxisPoints.forEach(function(item, index) {
```

Comment:
I think this would be clearer to read as a `map`

```JavaScript
$scope.throughput.xAxisPoints = $scope.endpoint.graphData.timeAxisValues.map(function(item) {
    var date = new Date(item);
    return date.toLocaleDateString() + ' ' + date.toLocaleTimeString();
});
```

That means the assignment 2 lines above can also be removed.

Reply:
Can potentially also remove the entire property as it's not used anywhere besides to create the xAxisPoints.

```JavaScript
delete $scope.endpoint.graphData.timeAxisValues;
```

But this is a purely superfluous suggestion.